corehq/apps/userreports/views.py

```diff
@@ -215,8 +216,26 @@ def page_context(self):
         return {
             'form': self.edit_form,
             'report': self.config,
+            'referring_apps': self.get_referring_apps(),
         }
 
+    def get_referring_apps(self):
```

Comment:
True. Is this a hard-ask on this PR? It's been sitting un-addressed altogether till now.

We could retire old builds, but I wonder if that would be an accepted product UX. One alternative is, to make a build of UCR source and report along with the app build and tie them together and not worry about deleting UCR sources. It aligns how we treat app build as a standalone artifact. Though, I could imagine it will add considerable size in case of large UCRs.

Reply:
Nope not at all a hard ask. Just a comment on how this won't actually confirm that a report is totally safe to delete

Reply:
This is a good point though, I will add a note on the dialogue to indicate the same.